test/parallel/test-async-local-storage-exit-does-not-leak.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+const common = require('../common');
+const { AsyncLocalStorage } = require('async_hooks');
+
+const als = new AsyncLocalStorage();
+
+// The als instance should be getting removed from the storageList in
+// lib/async_hooks.js when exit(...) is called, therefore when the nested runs
+// are called there should be no copy of the als in the storageList to run the
+// _propagate method on.
+als._propagate = common.mustNotCall('_propagate() should not be called');
```

Comment:
This is quite sensitive, e.g. someone renames `_propagate` and this test still passes but is worthless.
What about
```
const fName = "_propagate";
assert.ok(typeof(als[fName]) === "function");
als[fName] = common.mustNotCall('_propagate() should not be called');
```